### PR TITLE
Men 1151

### PR DESCRIPTION
**Pull request by oleorhagen**
Now handles the case where it would previously crash when no sys-certs were available on the system.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -191,22 +191,34 @@ func loadServerTrust(conf Config) (*x509.CertPool, error) {
 		return nil, nil
 	}
 
-	certs, err := x509.SystemCertPool()
+	syscerts, err := x509.SystemCertPool()
+	// fmt.Println(syscerts)
```
Comment (bboozzoo) [inline client/client.go]: fmt?

File: client/client.go [hunk 2]
```diff
@@ -191,22 +191,34 @@ func loadServerTrust(conf Config) (*x509.CertPool, error) {
...
 	if err != nil {
 		return nil, err
 	}
 
 	// Read certificate file.
-	cacert, err := ioutil.ReadFile(conf.ServerCert)
+	servcert, err := ioutil.ReadFile(conf.ServerCert)
+	// fmt.Println(servcert)
```
Comment (bboozzoo) [inline client/client.go]: fmt

File: client/client_test.go [hunk 3]
```diff
@@ -164,3 +167,29 @@ func TestCaLoading(t *testing.T) {
 	assert.True(t, systemOK)
 	assert.True(t, oursOK)
 }
+
+func TestEmptySystemCertPool(t *testing.T) {
+	version := runtime.Version()
+	if version == "1.6" || version == "1.7" || version == "1.8" {
```
Comment (bboozzoo) [inline client/client_test.go]: does 1.8.1 also return `"1.8"` runtime?

File: client/client.go [hunk 4]
```diff
@@ -191,22 +191,34 @@ func loadServerTrust(conf Config) (*x509.CertPool, error) {
...
 	if err != nil {
 		return nil, err
 	}
-	certs.AppendCertsFromPEM(cacert)
 
-	if len(certs.Subjects()) == 0 {
+	if servcert == nil && syscerts == nil {
```
Comment (bboozzoo) [inline client/client.go]: I'm guessing your intention is to handle a case when server certificat exists, but is empty, so `if len(servcert) == 0 { ...`  would be more appropriate.

You create a new cert pool a couple of lines down, so I don't see why checking `syscerts == nil` is needed here.

Comment (kacf) [inline client/client.go]: The intention is to catch when there is neither a specified certificate nor a pool of system certificates. I don't think we have handled the case where the file exists, but is empty. I'm not even sure if that would return nil or not. Might be worth checking.

**Comment by GregorioDiStefano** [comment]
Integrations tests aren't running because @oleorhagen  needs to be part of the mendersoftware github group/org

File: client/client.go [hunk 5]
```diff
@@ -191,22 +191,32 @@ func loadServerTrust(conf Config) (*x509.CertPool, error) {
...
 	// Read certificate file.
-	cacert, err := ioutil.ReadFile(conf.ServerCert)
+	servcert, err := ioutil.ReadFile(conf.ServerCert)
 	if err != nil {
 		return nil, err
 	}
-	certs.AppendCertsFromPEM(cacert)
 
-	if len(certs.Subjects()) == 0 {
+	if len(servcert) == 0 {
```
Comment (oleorhagen) [inline client/client.go]: Now doing prefix string search for version, and removed the commented out code. Also added @bboozzoo 
's suggestion of simply checking the number of bytes read from ReadFile.

File: client/client_test.go [hunk 6]
```diff
@@ -164,3 +167,29 @@ func TestCaLoading(t *testing.T) {
 	assert.True(t, systemOK)
 	assert.True(t, oursOK)
 }
+
+func TestEmptySystemCertPool(t *testing.T) {
+	version := runtime.Version()
+	if strings.HasPrefix(version, "1.6") || strings.HasPrefix(version, "1.7") || strings.HasPrefix(version, "1.8") {
```
Comment (oleorhagen) [inline client/client_test.go]: Now checks the prefix of the version string to handle 1.x.x etc.

File: client/client.go (hunk 5 above)
Comment (bboozzoo) [inline client/client.go]: yup, `len()` is smart enough to return 0 if `severcert == []byte{}` or `servercert == nil`

File: client/client_test.go [hunk 7]
```diff
@@ -164,3 +167,29 @@ func TestCaLoading(t *testing.T) {
...
+		// Environment variable not included until version 1.9. Therefore skipping this test.
+		t.SkipNow()
+	}
+
+	tmpdir, err := ioutil.TempDir("", "nocertsfolder")
+	assert.NoError(t, err)
+
+	// Fake the environment variables, to override ssl-cert lookup
+	err = os.Setenv("SSL_CERT_DIR", tmpdir)
```
Comment (bboozzoo) [inline client/client_test.go]: This is go 1.9 goodness, right?

Comment (oleorhagen) [inline client/client_test.go]: That is right ^^

**Comment by kacf** [comment]
Not sure why integration tests didn't run. Running this [internally](https://ci.cfengine.com/job/yoctobuild-pr/642).

**Pull request merged**